src/Relude/Extra/Type.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ExplicitForAll      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+
+{- |
+Copyright:  (c) 2018 Kowainik
+License:    MIT
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Contains useful functions to work with Types.
+-}
+
+module Relude.Extra.Type
+       ( typeName
+       ) where
+
+import Relude
+
+import Type.Reflection (Typeable, typeRep)
```

Comment:
You should use `Data.Typeable` to support GHC-8.0.2. As you noticed, there's no `Type.Reflection` in GHC-8.0.2. But `Data.Typeable` is implemented on top of `Type.Reflection` and always will be in `base` for backwards compatibility. You can read more about different `TypeRep`s in this blog post:

* https://kowainik.github.io/posts/2018-07-11-typerep-map-step-by-step#keys

Alternatively, it would be better if you can use `-XCPP` extension to use either `Type.Reflection` or `Data.Typeable` depending on the `base` version. It doesn't improve implementation much, but will help us to not forget to update all places when will drop support for GHC-8.0.2.